### PR TITLE
Setup a package for ODF extensions

### PR DESCRIPTION
**Pull request by bipuladh**
Refactors ODF Dashboard

Signed-off-by: Bipul Adhikari <badhikar@redhat.com>

Screenshot:
Screenshot from 2023-01-16 15-13-20

### CODE REVIEW

File: packages/odf/components/odf-dashboard/dashboard.tsx [hunk 1]
```diff
@@ -61,51 +64,52 @@ export const ODFDashboard: React.FC = () => {
...
+        let pushLocation = updatedPages.findIndex((pg) => pg.id === indexId);
+        if (extension.properties.before && updatedPages.length > 0) {
+          pushLocation -= 1;
+        }
+        if (updatedPages.length > 0) {
+          updatedPages.splice(pushLocation, 0, page);
+        } else {
+          updatedPages.push(page);
+        }
       });
```
Comment (SanjalKatiyar) [inline packages/odf/components/odf-dashboard/dashboard.tsx:107]: let's say I want to add before "overview":
pushLocation will be 0, then "pushLocation -= 1;" will make it -1,
nav will get added to the end instead in this case...

Comment (SanjalKatiyar) [reply packages/odf/components/odf-dashboard/dashboard.tsx:107]: similar issue can occur if we are searching (before/after) for any extension not yet in the list... in this case pushLocation will be  -1 which will get updated to  -2 and may cause unexpected result (depends)...

Comment (SanjalKatiyar) [reply packages/odf/components/odf-dashboard/dashboard.tsx:107]: it is internal use extension so may not be a big issue... still just pointing out once...

**Comment by SanjalKatiyar** [comment]
@bipuladh just to confirm once, did u test create/details page routes for noobaa resources as well ??
I think "Create" button on list page is still pointing to "/odf/resource/..."

**Comment by SanjalKatiyar** [comment]
> @bipuladh just to confirm once, did u test create/details page routes for noobaa resources as well ?? I think "Create" button on list page is still pointing to "/odf/resource/..."

same for breadcrumbs on details pages...

**Comment by bipuladh** [comment]
@SanjalKatiyar this PR is getting confusing I will send another PR with route changes for now reverting those changes.

**Pull request merged**